arch/sim/src/sim/up_devconsole.c

```diff
@@ -38,119 +38,245 @@
  ****************************************************************************/
 
 #include <nuttx/config.h>
+#include <nuttx/serial/serial.h>
 
-#include <sys/types.h>
-#include <stdbool.h>
-#include <sched.h>
-#include <errno.h>
-#include <fcntl.h>
+#include "up_internal.h"
 
-#include <nuttx/fs/fs.h>
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
 
-#include "up_internal.h"
+#define DEVCONSOLE_BUFSIZE 256
 
 /****************************************************************************
  * Private Function Prototypes
  ****************************************************************************/
 
-static ssize_t devconsole_read(FAR struct file *filep, FAR char *buffer,
-                 size_t buflen);
-static ssize_t devconsole_write(FAR struct file *filep,
-                 FAR const char *buffer, size_t buflen);
-static int     devconsole_poll(FAR struct file *filep, FAR struct pollfd *fds,
-                 bool setup);
+static int  devconsole_setup(FAR struct uart_dev_s *dev);
+static void devconsole_shutdown(FAR struct uart_dev_s *dev);
+static int  devconsole_attach(FAR struct uart_dev_s *dev);
+static void devconsole_detach(FAR struct uart_dev_s *dev);
+static int  devconsole_ioctl(FAR struct file *filep, int cmd, unsigned long arg);
+static int  devconsole_receive(FAR struct uart_dev_s *dev, uint32_t *status);
+static void devconsole_rxint(FAR struct uart_dev_s *dev, bool enable);
+static bool devconsole_rxavailable(FAR struct uart_dev_s *dev);
+static void devconsole_send(FAR struct uart_dev_s *dev, int ch);
+static void devconsole_txint(FAR struct uart_dev_s *dev, bool enable);
+static bool devconsole_txready(FAR struct uart_dev_s *dev);
+static bool devconsole_txempty(FAR struct uart_dev_s *dev);
 
 /****************************************************************************
  * Private Data
  ****************************************************************************/
 
-static const struct file_operations devconsole_fops =
+static const struct uart_ops_s g_devconsole_fops =
```

Comment:
maybe it's better to drop "f" from the name as it's no longer file_operations?

Reply:
Ok.